examples/ogbn_train.py

```diff
@@ -0,0 +1,266 @@
+)
+val_loader = NeighborLoader(
+    data,
+    input_nodes=split_idx['valid'],
+    num_neighbors=[args.fan_out] * num_layers,
+    batch_size=batch_size,
+    shuffle=True,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+test_loader = NeighborLoader(
+    data,
+    input_nodes=split_idx['test'],
+    num_neighbors=[args.fan_out] * num_layers,
+    batch_size=batch_size,
+    shuffle=True,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+
+
+def train(epoch: int) -> Tuple[Tensor, float]:
+    model.train()
+
+    pbar = tqdm(total=split_idx['train'].size(0))
+    pbar.set_description(f'Epoch {epoch:02d}')
+
+    total_loss = total_correct = 0
+    for batch in train_loader:
+        optimizer.zero_grad()
+        out = model(batch.x, batch.edge_index.to(device))[:batch.batch_size]
+        y = batch.y[:batch.batch_size].squeeze().to(torch.long)
+        loss = F.cross_entropy(out, y)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss)
+        total_correct += int(out.argmax(dim=-1).eq(y).sum())
+        pbar.update(batch.batch_size)
+
+    pbar.close()
+    loss = total_loss / len(train_loader)
+    approx_acc = total_correct / split_idx['train'].size(0)
+    return loss, approx_acc
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, val_steps=None) -> float:
+    model.eval()
+
+    total_correct = total_examples = 0
+    for i, batch in enumerate(loader):
+        if val_steps is not None and i >= val_steps:
+            break
+        batch = batch.to(device)
+        batch_size = batch.num_sampled_nodes[0]
+        out = model(batch.x, batch.edge_index)[:batch_size]
+        pred = out.argmax(dim=-1)
+        y = batch.y[:batch_size].view(-1).to(torch.long)
+
+        total_correct += int((pred == y).sum())
+        total_examples += y.size(0)
+
+    return total_correct / total_examples
+
+
+if args.use_gat:
+    from torch_geometric.nn.models import GAT
+    model = GAT(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+        heads=args.num_heads,
+    )
+else:
+    from torch_geometric.nn.models import GraphSAGE
+    model = GraphSAGE(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+    )
+
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                             weight_decay=args.wd)
+
+if verbose:
+    prep_time = round(time.perf_counter() - wall_clock_start, 2)
+    print("Total time before training begins (prep_time)=", prep_time,
+          "seconds")
+    print("Training...")
+
+test_accs = []
+val_accs = []
+times = []
+train_times = []
+inference_times = []
+best_val = best_test = 0.
+start = time.time()
```

Comment:
nit: Can we use `time.perf_counter()` for consistency in the script?
```suggestion
start = time.perf_counter()
```